tests/unit/test_utils.py

```diff
@@ -1,9 +1,236 @@
+    half_simple_df.data["value"].loc[0] = 0.49
+    scenarios = find_matching_scenarios(
+        simple_df,
+        half_simple_df,
+        variable_follower,
+        variable_leaders,
+        ["scen_a", "scen_b"],
+    )
+    assert scenarios == ("*", "scen_a")
+    half_simple_df.data["value"].loc[0] = 0.51
+    scenarios = find_matching_scenarios(
+        simple_df,
+        half_simple_df,
+        variable_follower,
+        variable_leaders,
+        ["scen_a", "scen_b"],
+    )
+    assert scenarios == ("*", "scen_b")
+
+
+def test_find_matching_scenarios_no_data_for_time():
+    variable_leaders = ["Emissions|CO2"]
+    variable_follower = "Emissions|CH4"
+    time_col = simple_df.time_col
+    half_simple_df = simple_df.filter(scenario="scen_a")
+    half_simple_df.data[time_col].loc[0] = 0
+    with pytest.raises(ValueError):
+        find_matching_scenarios(
+            simple_df,
+            half_simple_df,
+            variable_follower,
+            variable_leaders,
+            ["scen_a", "scen_b"],
+        )
+
+
+def test_find_matching_scenarios_use_change_instead_of_absolute():
+    # In this case, we will ignore any offset
+    variable_leaders = ["Emissions|CO2"]
+    variable_follower = "Emissions|CH4"
+    half_simple_df = simple_df.filter(scenario="scen_a")
+    half_simple_df.data["value"] = half_simple_df.data["value"] - 10000
+    scenarios = find_matching_scenarios(
+        simple_df,
+        half_simple_df,
+        variable_follower,
+        variable_leaders,
+        ["scen_a", "scen_b"],
+        use_change_not_abs=True,
+    )
+    assert scenarios == ("*", "scen_a")
+
+
+def test_find_matching_scenarios_complicated():
+    # This is similar to the above case except with multiple models involved and
+    # requiring specific interpolation. First we construct the database:
+    variable_leaders = ["Emissions|CO2"]
+    variable_follower = "Emissions|CH4"
+    df_low = simple_df.copy()
+    df_low.data["scenario"].loc[
+        df_low.data["scenario"] == "scen_a"
+        ] = "right_scenario"
+    df_low.data["scenario"].loc[
+        df_low.data["scenario"] == "scen_b"
+        ] = "wrong_scenario"
+    df_high = df_low.copy()
+    df_high["model"] = "high_model"
+    df_low.data["value"] = df_low.data["value"] - 10
+    df_high.data["value"] = df_high.data["value"] + 11
+    df_near = simple_df.copy()
+    df_near.data["value"] = df_near["value"] + 1
+    df_to_test = df_low.append(df_near).append(df_high)
+    # We need to refresh the metadata in order to proceed.
+    df_to_test = pyam.IamDataFrame(df_to_test.data)
+
+    # Now the tests can begin
+    scenario = find_matching_scenarios(
+        df_to_test,
+        simple_df,
+        variable_follower,
+        variable_leaders,
+        ["scen_a", "scen_b", "right_scenario"],
+    )
+    assert scenario == ("*", "right_scenario")
+    # If the "wrong scenario" option is passed in as an earlier option, it will be
+    # selected instead
```

Comment:
because an average over all the models is being done?

Reply:
Yes, so it's a draw between them